### PR TITLE
split the volume size field into two

### PR DESCRIPTION
**Pull request by ChengYanJin**
**Component**: ui, storage

**Context**:  We would like to split the volume size into two. One is input for the positive integer, the other is select for the unit.

**Summary**: For the input field, we check if it's the positive integer. And users can select the unit from the dropdown list.

**Acceptance criteria**: 
image

See: #1459

### CODE REVIEW

File: ui/src/services/utils.js [hunk 1]
```diff
@@ -98,3 +98,11 @@ export const useRefreshEffect = (refreshAction, stopRefreshAction) => {
     };
   }, [dispatch, refreshAction, stopRefreshAction]);
 };
+
+export const sizeUnits = [
+  { label: 'ki', value: 'Ki' },
```
Comment (slaperche-scality) [inline ui/src/services/utils.js]: `ki` => `KiB`

I know that I told you that it was a small `k`, but actually it's more complicated:
- a small `k` for [kilobyte](https://en.wikipedia.org/wiki/Kilobyte) (`kB`) 
- a big `K` for [kibibyte](https://en.wikipedia.org/wiki/Kibibyte) (`KiB`)

File: ui/src/services/utils.js [hunk 2]
```diff
@@ -98,3 +98,11 @@ export const useRefreshEffect = (refreshAction, stopRefreshAction) => {
     };
   }, [dispatch, refreshAction, stopRefreshAction]);
 };
+
+export const sizeUnits = [
```
Comment (slaperche-scality) [inline ui/src/services/utils.js]: I would add the `B` of byte for the labels, so: `KiB`, `MiB`, …

Also, I would add `{ label: 'B', value: '' }` as one may want to specify the size in bytes directly.

File: ui/src/translations/en.json [hunk 3]
```diff
@@ -87,5 +87,5 @@
   "volume_creation_failed": "Volume {name} creation has failed.",
   "volume_name_error": "Name must be lowercase letters, numbers and hyphens.",
   "volume_path_error": "Device path must start with a '/'",
-  "volume_size_error": "Size must be composed of a positive integer and a unit like 250Gi or 1Ti."
+  "volume_size_error": "The size must be a postive integer."
```
Comment (slaperche-scality) [inline ui/src/translations/en.json]: `postive` => `positive`

File: ui/src/containers/CreateVolume.js [hunk 4]
```diff
@@ -88,37 +101,44 @@ const CreateVolume = props => {
...
-      )
-      .when('type', {
-        is: SPARSE_LOOP_DEVICE,
-        then: Yup.string().required()
-      })
+    sizeInput: yup.number().when('type', {
+      is: SPARSE_LOOP_DEVICE,
+      then: yup
+        .number()
+        .integer(intl.messages.volume_size_error)
```
Comment (slaperche-scality) [inline ui/src/containers/CreateVolume.js]: This is way more readable than a regex :+1: 

That being said, it's buggy :disappointed: 

If you input a "big" number, say 2 GiB as bytes (which is pretty small actually), e.g. `2147483648`, you have the error `The size must be a postive integer.` which is pretty weird from a user point of view (it is a positive integer).

I guess we wouldn't have the problem with the good ol’ regex :stuck_out_tongue:

**Review by slaperche-scality — CHANGES_REQUESTED** [review]

File: ui/src/containers/CreateVolume.js [hunk 5]
```diff
@@ -127,17 +127,16 @@ const CreateVolume = props => {
...
       }),
-    sizeInput: yup.number().when('type', {
+    sizeInput: yup.string().when('type', {
       is: SPARSE_LOOP_DEVICE,
       then: yup
-        .number()
-        .integer(intl.messages.volume_size_error)
-        .positive(intl.messages.volume_size_error)
+        .string()
+        .matches(/^[1-9][0-9]*$/, intl.messages.volume_size_error)
```
Comment (slaperche-scality) [inline ui/src/containers/CreateVolume.js]: It would be nice to explain in the commit message why we went back to the regex

**Review by slaperche-scality — COMMENTED** [review]
Works fine :smiley: 

Two remarks though:

1. Not in the scope of this PR but I realized that we're not checking if the volume name is unique.
2. I have the error message `sizeInput is a required field` in English, even in French (and `sizeInput` may not be very clear to the user). I guess it's an error message generated by `Yup`, I don't know if we can override it (if yes, we should do it).

**Comment by carlito-scality** [comment]
Select Input style should be improved to be consistent with other inputs but can be managed later.

**Review by slaperche-scality — DISMISSED** [review]

**Comment by ChengYanJin** [comment]
@bert-e approve

**Review by MonPote — APPROVED** [review]

**Pull request merged**